endorser/services/endorsement_processor.py

```diff
@@ -79,150 +67,41 @@ def are_tasks_running(self) -> bool:
-            list_key: The Redis key of the list to process.
-        """
-        logger.trace("Attempt process: {}", event_key)
-        lock_key = f"lock:{event_key}"
-        extend_lock_task = None
-
-        lock_duration = 1  # second
-
-        if self.redis_service.set_lock(
-            key=lock_key,
-            px=lock_duration * 1000,  # to milliseconds
-        ):
-            logger.trace("Successfully set lock for {}", event_key)
-            event_json = self.redis_service.get(event_key)
-            if not event_json:
-                logger.warning(
-                    "Tried to read an event from key {}, but event has been deleted:",
-                    event_key,
-                )
-                return
-
-            try:
-                # Start a background task to extend the lock periodically
-                extend_lock_task = self.redis_service.extend_lock_task(
-                    lock_key, interval=datetime.timedelta(seconds=lock_duration)
-                )
-
-                await self._process_endorsement_event(event_json)
-                if self.redis_service.delete_key(event_key):
-                    logger.info("Deleted processed endorsement event: {}", event_key)
-                else:
-                    logger.warning(
-                        "Couldn't delete processed endorsement event: {}", event_key
+                messages = await subscription.fetch(10, 1)
+                for message in messages:
+                    message_subject = message.subject
+                    message_data = message.data.decode()
+                    logger.trace(
+                        "received message: {} with subject {}",
+                        message_data,
+                        message_subject,
                     )
-            except Exception as e:  # pylint: disable=W0718
-                # if this particular event is unprocessable, we should remove it from the inputs, to avoid deadlocking
-                logger.error("Processing {} raised an exception: {}", event_key, e)
-                self._handle_unprocessable_endorse_event(event_key, event_json, e)
-            finally:
-                # Cancel the lock extension task if it's still running
-                if extend_lock_task:
-                    extend_lock_task.cancel()
-        else:
-            logger.debug(
-                "Event {} is currently being processed by another instance.", event_key
-            )
+                    await self._process_endorsement_event(message_data)
+                    await message.ack()
+            except TimeoutError:
+                logger.trace("Timeout fetching messages continuing...")
+                await asyncio.sleep(1)
+            except Exception as e:  # pylint: disable=W0703
+                logger.error("Error processing endorsement event: {}", e)
+                await self._handle_unprocessable_endorse_event(
+                    message_subject, message_data, e
+                )
```

Comment:
Technically not safe... because message_subject is potentially unbound.

e.g. endorser just starts up, `message` has not been defined yet, `subscription.fetch` raises some exception (other than TimeoutError), and then line 102 will raise a ValueError, var not defined. It's good to look out for these possibilities in try-except blocks, where new vars are defined in the try, and then used in the exception handling.